CMakeLists.txt

```diff
@@ -145,7 +145,9 @@ if(WITH_ZMQ)
     find_package(ZeroMQ CONFIG REQUIRED)
   else()
     # The ZeroMQ project has provided config files since v4.2.2.
-    # TODO: Switch to find_package(ZeroMQ) at some point in the future.
+    # However, mainstream distributions do not yet provide CMake
```

Comment:
It's disappointing that so many distros don't support CMake in their packaging. We basically can't switch this (and any other packages where we might want to use the "native" CMake config) until *every* distro does, otherwise it'll just be broken for that distro.

Some do, i.e [Alpine](https://pkgs.alpinelinux.org/contents?branch=edge&name=zeromq%2ddev&arch=x86&repo=main), or [Arch](https://archlinux.org/packages/extra/x86_64/zeromq/), or [nix](https://github.com/NixOS/nixpkgs/blob/release-23.11/pkgs/development/libraries/zeromq/4.x.nix), but the majority do not, ie Debian, Ubuntu, Fedora, *BSDs, as well as other package managers, i.e `brew` or `Guix`. Gentoo has had an issue open for adding CMake support for ~5 years: https://bugs.gentoo.org/695202, but no change.

There are multiple issues here (and it's not really all the fault of the distros):
* One is that distros won't switch build systems if the current one works fine, and switching doesn't provide (m)any benefits, while having a 100% chance of breaking dependant packages/other things.
* Another is that packages like zeromq, which have opted to provide two build systems, don't properly support installing the config files from one, in the other. 
  * i.e if you build with CMake you'll get CMake config files, and pkg-config files. However if you build with Autotools, you just get pkg-config files.
  * One solution to try and speed up Cmake adoption wouuld be to upstream support for installing CMake config files from Autotools.

I assume one thing we could do, is try `find_package(ZeroMQ)`  first, and fallback to `pkg_check_modules(libzmq)`, and just maintain this (pretty much) forever. That might be better than somewhat vauge comments about "modern" distros, because, (some) modern distros do already support this, and it's actually the case that *all* versions of *all* distros Bitcoin Core supports need to do this, before we could do a complete switch.

Reply:
This explanation makes more sense to me than the current code comment.

> try `find_package(ZeroMQ)` first, and fallback to `pkg_check_modules(libzmq)`

That would be nice. Then a short clarifying comment could be:

```
# Use CMake config file if the distro provides it (pseudo code)
try find_package(ZeroMQ)
# Otherwise 
catch pkg_check_modules(libzmq)
```

Reply:
Discussed offline, and we will change the code to try both in a follow up.

Reply:
Opened an issue for this #30876.